src/test/java/sysc4806/graduateAdmissions/controller/RoleControllerTest.java

```diff
@@ -0,0 +1,139 @@
+        Set<Privilege> adminPrivileges = new HashSet<>();
+        studentPrivileges.add(Privilege.builder().id(1L).owner(Owner.SELF).build());
+        Set<Privilege> professorPrivileges = new HashSet<>();
+        studentPrivileges.add(Privilege.builder().id(2L).owner(Owner.ALL_PROFS).build());
+
+        roles = Arrays.asList(
+                Role.builder().roleName("Student").privileges(studentPrivileges).build(),
+                Role.builder().roleName("Admin").privileges(adminPrivileges).build(),
+                Role.builder().roleName("Professor").privileges(professorPrivileges).build());
+
+        when(repository.findAll()).thenReturn(roles);
+        when(repository.findByRoleName("Admin")).thenReturn(Optional.of(roles.get(1)));
+        when(repository.findByRoleName("Student")).thenReturn(Optional.of(roles.get(0)));
+    }
+
+    /* Test get roles without id */
+    @Test
+    public void testGetAllRoles() throws Exception {
+        this.mockMvc.perform(get("/role/"))
+                .andExpect(status().isOk())
+                .andExpect(content().json(toJson(roles)));
+    }
+
+    /* Test getting role by name */
+    @Test
+    public void testGetRoleByName() throws Exception {
+        this.mockMvc.perform(get("/role/?name=Admin"))
+                .andExpect(status().isOk())
+                .andExpect(content().json(toJson(roles.get(1))));
+    }
+
+    /* Test create Role */
+    @Test
+    public void testCreateRole() throws Exception {
+        Set<Privilege> privileges = new HashSet<>();
+        privileges.add(Privilege.builder().id(0L).owner(Owner.ALL_STUDENTS).build());
+        MvcResult result = mockMvc.perform(post("/role/create").contentType(APPLICATION_JSON_UTF8)
+                .content(toJson(
+                        Role.builder().roleName("new").privileges(privileges).build())))
+                .andExpect(status().isOk())
+                .andReturn();
+        assertTrue(result.getResponse().getContentAsString().contains("Role added"));
+    }
+
+    /* Test delete existing Role */
+    @Test
+    public void testDeleteRole() throws Exception {
+        MvcResult result = mockMvc.perform(delete("/role/{name}", "Student"))
+                .andExpect(status().isOk())
+                .andReturn();
+        assertTrue(result.getResponse().getContentAsString().contains("Student Role deleted"));
+    }
+
+    /* Test delete Role that does not exist */
+    @Test
+    public void testDeleteRoleDoesNotExist() throws Exception {
+        mockMvc.perform(delete("/role/{name}", "banana"))
+                .andExpect(status().isNotFound());
+    }
+
+    /* Test update Role */
+    @Test
+    public void testUpdateRole() throws Exception {
+        mockMvc.perform(post("/role/update").contentType(APPLICATION_JSON_UTF8)
+                .content(toJson(
+                        Role.builder().roleName("Student").build())))
+                .andExpect(status().isOk());
+    }
```

Comment:
What is the difference between this update role and the create role? (I actually did mine wrong in my PR too). The controller has two of the same methods but different endpoint and return string, but does the exact same thing.
[This article did a good job of explaining how the code should look](http://progressivecoder.com/rest-api-using-spring-boot-insert-update/) (minus the service layer what we have yet to implement.)
TL;DR:
|                 | PostMapping | PutMapping                                                     |   |   |
|-----------------|-------------|----------------------------------------------------------------|---|---|
| Returns         | 201 CREATED | 200 OK                                                         |   |   |
| Takes path var? | No          | Yes (xx/{id})                                                  |   |   |
| Code on Error   | 5xx series  | 404 when xx.id<br>not found.<br>(use existsById)<br>5xx series |   |   |